### PR TITLE
[SYCL][ESIMD]Implement gather(lacc) accepting compile time properties

### PR DESCRIPTION
**Pull request by fineg74**
This implements the unified memory API for gather with local accessors

### CODE REVIEW

File: sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp [hunk 1]
```diff
@@ -761,3 +761,254 @@ template <typename T, TestFeatures Features> bool testACC(queue Q) {
...
+       CGH.parallel_for(Range, [=](sycl::nd_item<1> NDI) SYCL_ESIMD_KERNEL {
+         uint16_t GlobalID = NDI.get_global_id(0);
+         uint16_t LocalID = NDI.get_local_id(0);
+         uint32_t GlobalElemOffset = GlobalID * N;
+         uint32_t LocalElemOffset = LocalID * N;
+
+         if (LocalID == 0) {
+           for (int I = 0; I < Threads * N; I += 8) {
+             simd<T, 8> InVec(In + GlobalElemOffset + I);
+             simd<uint32_t, 8> offsets(I * sizeof(T), sizeof(T));
```
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp:815]: nit: if use `SomeVar` style, then
```suggestion
             simd<uint32_t, 8> Offsets(I * sizeof(T), sizeof(T));
```

**Review by v-klochkov — COMMENTED** [review]
Looks really good! I cannot add anything except fixing 1 letter ))
@sarnex - please also take a look, I could miss something.

Assume approved, but pending to review/approval from Nick. Thank you.

File: sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp (hunk 1 above)
Comment (v-klochkov) [reply sycl/test-e2e/ESIMD/unified_memory_api/Inputs/gather.hpp:815]: +1 more comment: Please ensure there are calls and compile-tests like: 
gather(lacc, byte_offsets, 0), and gather(lacc, byte_offsets, 0, mask)
to verify lacc-ga-4,5  having MaskT template parameter is sufficient work-around for 2 older functions having 'glob_offsets' parameter.

**Review by sarnex — APPROVED** [review]
Really sorry, I promised I'd do this yesterday but something came up.

LGTM with Slava's comments

**Review by sarnex — APPROVED** [review]

File: sycl/test/esimd/memory_properties.cpp [hunk 2]
```diff
@@ -1218,6 +1218,12 @@ test_gather_scatter(AccType &acc, LocalAccType &local_acc, float *ptrf,
                                  pass_thru_view, props_align4);
   acc_res = gather<float, 32, 2>(local_acc, ioffset_n16_view, mask_n16,
                                  pass_thru_view, props_align4);
+
+  // Validate that a new API doesn't conflict with the new API.
```
Comment (v-klochkov) [inline sycl/test/esimd/memory_properties.cpp:1222]: nit:
```suggestion
  // Validate that a new API doesn't conflict with the old API.
```

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**